get_github_coderabbit_review/coderabbit_fetcher/async_processing/async_github_client.py

```
@@ -0,0 +1,499 @@
+                if response.status == 403 and "rate limit" in (await response.text()).lower():
+                    logger.warning("GitHub API rate limit exceeded")
+                    # Calculate wait time until rate limit resets
+                    wait_time = max(0, self.rate_limit_reset - time.time())
+                    if wait_time > 0 and wait_time < 3600:  # Wait up to 1 hour
+                        logger.info(f"Waiting {wait_time:.0f} seconds for rate limit reset")
+                        await asyncio.sleep(wait_time)
+                        # Retry the request
+                        return await self._make_request(method, url, **kwargs)
+                    raise APIRateLimitError(
+                        f"Rate limit exceeded. Reset at {self.rate_limit_reset}"
+                    )
```

Ruff (0.13.2)

Abstract `raise` to an inner function

(TRY301)